virttest/xml_utils.py

```diff
@@ -70,11 +70,9 @@ def __init__(self, suffix=TMPSFX, prefix=TMPPFX, mode="w+", buffsz=1):
         param: mode: second parameter to file()/open()
         param: buffer: third parameter to file()/open()
         """
-        self.fd, self.path = tempfile.mkstemp(suffix=suffix, prefix=prefix)
-        os.close(self.fd)
-        self.open_file = open(self.path, mode, buffsz)
-        self.name = self.open_file.name
-        super(TempXMLFile, self).__init__()
+        fd, path = tempfile.mkstemp(suffix=suffix, prefix=prefix)
```

Comment:
This `tempfile.mkstemp` is reported as insecurity, so I do not recommend we use it any more. Could you consider another one? I find in python document site below. 

```
tempfile.mktemp(suffix='', prefix='tmp', dir=None)
Deprecated since version 2.3: Use [mkstemp()](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp) instead.
```